tests/components/zha/test_channels.py

```diff
@@ -159,4 +171,203 @@ def test_channel_registry():
+@mock.patch("homeassistant.components.zha.core.channels.ChannelPool.add_relay_channels")
+@mock.patch(
+    "homeassistant.components.zha.core.discovery.PROBE.discover_entities",
+    mock.MagicMock(),
+)
+def test_ep_channels_all_channels(m1, zha_device_mock):
+    """Test EndpointChannels adding all channels."""
+    zha_device = zha_device_mock(
+        {
+            1: {"in_clusters": [0, 1, 6, 8], "out_clusters": [], "device_type": 0x0000},
+            2: {
+                "in_clusters": [0, 1, 6, 8, 768],
+                "out_clusters": [],
+                "device_type": 0x0000,
+            },
+        }
+    )
+    channels = zha_channels.Channels(zha_device)
+
+    ep_channels = zha_channels.ChannelPool.new(channels, 1)
+    assert "1:0x0000" in ep_channels.all_channels
+    assert "1:0x0001" in ep_channels.all_channels
+    assert "1:0x0006" in ep_channels.all_channels
+    assert "1:0x0008" in ep_channels.all_channels
+    assert "1:0x0300" not in ep_channels.all_channels
+    assert "2:0x0000" not in ep_channels.all_channels
+    assert "2:0x0001" not in ep_channels.all_channels
+    assert "2:0x0006" not in ep_channels.all_channels
+    assert "2:0x0008" not in ep_channels.all_channels
+    assert "2:0x0300" not in ep_channels.all_channels
+
+    channels = zha_channels.Channels(zha_device)
+    ep_channels = zha_channels.ChannelPool.new(channels, 2)
+    assert "1:0x0000" not in ep_channels.all_channels
+    assert "1:0x0001" not in ep_channels.all_channels
+    assert "1:0x0006" not in ep_channels.all_channels
+    assert "1:0x0008" not in ep_channels.all_channels
+    assert "1:0x0300" not in ep_channels.all_channels
+    assert "2:0x0000" in ep_channels.all_channels
+    assert "2:0x0001" in ep_channels.all_channels
+    assert "2:0x0006" in ep_channels.all_channels
+    assert "2:0x0008" in ep_channels.all_channels
+    assert "2:0x0300" in ep_channels.all_channels
+
+
+@mock.patch("homeassistant.components.zha.core.channels.ChannelPool.add_relay_channels")
+@mock.patch(
+    "homeassistant.components.zha.core.discovery.PROBE.discover_entities",
+    mock.MagicMock(),
+)
+def test_channel_power_config(m1, zha_device_mock):
+    """Test that channels only get a single power channel."""
+    in_clusters = [0, 1, 6, 8]
+    zha_device = zha_device_mock(
+        {
+            1: {"in_clusters": in_clusters, "out_clusters": [], "device_type": 0x0000},
+            2: {
+                "in_clusters": [*in_clusters, 768],
+                "out_clusters": [],
+                "device_type": 0x0000,
+            },
+        }
+    )
+    channels = zha_channels.Channels.new(zha_device)
+    pools = {pool.id: pool for pool in channels.pools}
+    assert "1:0x0000" in pools[1].all_channels
+    assert "1:0x0001" in pools[1].all_channels
+    assert "1:0x0006" in pools[1].all_channels
+    assert "1:0x0008" in pools[1].all_channels
+    assert "1:0x0300" not in pools[1].all_channels
+    assert "2:0x0000" in pools[2].all_channels
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5051600)